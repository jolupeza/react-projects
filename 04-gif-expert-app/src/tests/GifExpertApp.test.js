import { shallow } from 'enzyme';
import GiftExpertApp from '../GifExpertApp';

describe('Test component GifExpertApp', () => {
  test('should show component GifGrid', () => {
    const wrapper = shallow(<GiftExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show categories list', () => {
    const categories = ['One Punch', 'Dragin Ball'];
    const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});