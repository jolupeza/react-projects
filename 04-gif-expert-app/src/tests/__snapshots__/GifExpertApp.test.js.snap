// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component GifExpertApp should show categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragin Ball"
      key="Dragin Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test component GifExpertApp should show component GifGrid 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
